### DISCUSSION TITLE
[Fixed] VS2022 - Redo doesn't work when the Copilot extension is enabled

### DISCUSSION THREAD

**Discussion by Kimbatt** [community/community#14027] [Copilot Conversations] [answered]
*EDIT*: Seems to be fixed in version 1.12

**Description**:
When the Copilot extension is enabled in VS2022, the 'Redo' functionality of the text editor doesn't work (neither with clicking it in the Edit -> Redo menu, nor using Ctrl-Y hotkey).
Changing the extension settings didn't seem to do anything, only disabling the extension fixed the problem for me.

**Steps to reproduce**:
Open a project with VS2022 with the Copilot extension enabled, modify the text, undo the modifications, then try redoing it.

**Expected behavior**:
The Redo functionality should work when the extension is enabled.

I used:
Visual Studio Community 2022 (64-bit), Version 17.1.3
Copilot version 1.10.0.3

Video:
https://user-images.githubusercontent.com/29901058/161228129-93ff4760-35ed-4d9b-82ca-2685d1855e74.mp4

**Comment by goravine** [comment]
I am also having the same problem, 
already reinstalling VS 2022 but redo not working when copilot extension enabled

**Comment by FuSioN79m** [comment]
Me too

**Comment by SordicSoftware** [comment]
This is an issue with the most recent version of Copilot. This is an extremely breaking issue

**Comment by sinloss** [comment]
Same here

**Comment by NathanBarnard** [comment]
I'm experiencing this too.

**Comment by Red007Master** [comment]
same

**Comment by Carandiru0** [comment]
same here, vs 2022, redo does not work! undo works....

**Comment by Henderv** [comment]
#14115

**Comment by silmoonsone** [comment]
我也有同样的问题，希望给予修复。

**Comment by chipnertkj** [comment]
I have a theory as to why this is happening. I've noticed that copilot sometimes adds and instantly removes a tabulation character(s). There are cases where it fails to remove that character(s) when you try to quickly type in something when adding a new line - that's how I noticed this in the first place. These modifications might be what's causing the undo history to get cleared.

**Reply by FuSioN79m** [reply]
> I have a theory as to why this is happening. I've noticed that copilot sometimes adds and instantly removes a tabulation character(s). There are cases where it fails to remove that character(s) when you try to quickly type in something when adding a new line - that's how I noticed this in the first place. These modifications might be what's causing the undo history to get cleared.

It's a good theory!

**Reply by FireToDust** [reply]
Probably not, because it still shows the undo history (at least for me) if you click the down arrow next to redo, just doesn't do anything when you click any of them.

**Comment by quico14** [comment]
Same for me :(

**Comment by xuetongoll** [comment]
same :(

**Comment by jangofett890** [comment]
I have the same issue still.

**Comment by FuSioN79m** [comment, marked as answer]
Update
GitHub Copilot Agent v1.12.0
Seems to be fixed

**Comment by kzu** [comment]
This is indeed fixed in v1.12

**Comment by beingminimal** [comment]
Hey everyone please join discussion here and vote also here. It's diff topic but requires attention to improve DX.
https://github.com/github/feedback/discussions/15533
Thanks
